src/ui/public/filter_bar/filter_bar.js

```diff
@@ -28,13 +29,35 @@ define(function (require) {
           'invertFilter',
           'invertAll',
           'removeFilter',
-          'removeAll'
+          'removeAll',
+          'updateFilter'
         ].forEach(function (method) {
           $scope[method] = queryFilter[method];
         });
 
         $scope.state = getAppState();
 
+        $scope.aceLoaded = function (editor) {
+          editor.$blockScrolling = Infinity;
+          var session = editor.getSession();
+          session.setTabSize(2);
+          session.setUseSoftTabs(true);
+
+          session.on('changeAnnotation', function checkIfValid() {
+            var annotations = session.getAnnotations();
+            $scope.editingFilterError = _.some(annotations, {type: 'error'});
```

Comment:
Can we move this to use `ngModelController.$setValidity()` so that angular knows the validity state of the editor, which can be used in many of the same ways this is but also extends to multiple inputs when necessary.